File: src/main/java/net/cjsah/console/exceptions/Para1CommandException.java
package net.cjsah.console.exceptions;

import net.cjsah.console.command.StringReader;

import java.util.function.Function;

public class Para1CommandException implements CommandExceptionType {
    private final Function<Object, String> function;

    public Para1CommandException(final Function<Object, String> function) {
        this.function = function;
    }

    public CommandException create(Object arg) {
        return new CommandException(this, this.function.apply(arg));
    }

    public CommandException createWithContext(StringReader reader, Object arg) {
        return new CommandException(this, this.function.apply(arg), reader.getString(), reader.getCursor());
    }


}

File: src/main/kotlin/net/cjsah/console/exceptions/Para1CommandException.kt
@file:Suppress("unused")

package net.cjsah.console.exceptions

import net.cjsah.console.command.StringReader
import java.util.function.Function

class Para1CommandException(private val function: Function<Any, String>) : CommandExceptionType {

    fun create(arg: Any) = CommandException(this, function.apply(arg))

    fun createWithContext(reader: StringReader, arg: Any) = CommandException(this, function.apply(arg), reader.getString(), reader.getCursor())

}